cli/cdsctl/environment_variable.go

```diff
@@ -0,0 +1,98 @@
+package main
+
+import (
+	"github.com/spf13/cobra"
+
+	"github.com/ovh/cds/cli"
+	"github.com/ovh/cds/sdk"
+)
+
+var (
+	environmentVariableCmd = cli.Command{
+		Name:  "variable",
+		Short: "Manage CDS environment variables",
+	}
+
+	environmentVariable = cli.NewCommand(environmentVariableCmd, nil,
+		[]*cobra.Command{
+			cli.NewCommand(environmentVariableCreateCmd, environmentCreateVariableRun, nil),
+			cli.NewListCommand(environmentVariableListCmd, environmentListVariableRun, nil),
+			cli.NewCommand(environmentVariableDeleteCmd, environmentDeleteVariableRun, nil),
+			cli.NewCommand(environmentVariableUpdateCmd, environmentUpdateVariableRun, nil),
+		})
+)
+
+var environmentVariableCreateCmd = cli.Command{
+	Name:  "add",
+	Short: "Add a new variable on environment. variable type can be one of password, text, string, key, boolean, number, repository",
+	Args: []cli.Arg{
+		{Name: "project-key"},
+		{Name: "env-name"},
+		{Name: "variable-name"},
+		{Name: "variable-type"},
+		{Name: "variable-value"},
+	},
+}
+
+func environmentCreateVariableRun(v cli.Values) error {
+	variable := &sdk.Variable{
+		Name:  v["variable-name"],
+		Type:  v["variable-type"],
+		Value: v["variable-value"],
+	}
+	if err := client.EnvironmentVariableCreate(v["project-key"], v["env-name"], variable); err != nil {
+		return err
+	}
+	return nil
+}
+
+var environmentVariableListCmd = cli.Command{
+	Name:  "list",
+	Short: "List CDS environment variables",
+	Args: []cli.Arg{
+		{Name: "project-key"},
+		{Name: "env-name"},
+	},
+}
+
+func environmentListVariableRun(v cli.Values) (cli.ListResult, error) {
+	variables, err := client.EnvironmentVariablesList(v["project-key"], v["env-name"])
+	if err != nil {
+		return nil, err
+	}
+	return cli.AsListResult(variables), nil
+}
+
+var environmentVariableDeleteCmd = cli.Command{
+	Name:  "delete",
+	Short: "Delete CDS environment variable",
+	Args: []cli.Arg{
+		{Name: "project-key"},
+		{Name: "env-name"},
+		{Name: "variable-name"},
+	},
+}
+
+func environmentDeleteVariableRun(v cli.Values) error {
+	return client.EnvironmentVariableDelete(v["project-key"], v["env-name"], v["variable-name"])
+}
+
+var environmentVariableUpdateCmd = cli.Command{
+	Name:  "update",
+	Short: "Update CDS environment variable value",
+	Args: []cli.Arg{
+		{Name: "project-key"},
+		{Name: "env-name"},
+		{Name: "variable-name"},
+		{Name: "variable-value"},
+	},
+}
+
+func environmentUpdateVariableRun(v cli.Values) error {
+	variable, err := client.EnvironmentVariableGet(v["project-key"], v["env-name"], v["variable-name"])
```

Comment:
You will need old variable name to be able to rename the variable
update name/value and type